deltachat-ios/Helper/TimeBucket.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+
+enum TimeBucket: CaseIterable {
+    case today
+    case yesterday
+    case thisWeek
+    case lastWeek
+    case thisMonth
+    case lastMonth
+}
+
+extension TimeBucket {
+
+    static func bucket(for date: Date) -> TimeBucket {
+
+        if date.isInToday {
+            return .today
+        }
+
+        if date.isInYesterday {
+            return .yesterday
+        }
+        if date.isInThisWeek {
+            return .thisWeek
+        }
+
+        if date.isInLastWeek {
+            return .lastWeek
+        }
+
+        if date.isInThisMonth {
+            return .thisMonth
+        }
+
+        if date.isInLastMonth {
+            return .lastMonth
+        }
+
+        // TODO: handle months
+        return .lastMonth
```

Comment:
this probably needs another enum value

Reply:
> Hey @nayooti, the gallery already looks really nice. Some nitpicking from my side:
> 
> * The section header of could have a light gray background in order to differentiate the time buckets better. Personally I would also prefer to have not capital letters, but I think this is an iOS convention.

Solved by changing the concept. See PR-description update.

> * There's an empty state for the gallery missing. Iirc @r10s implemented some weeks ago a check that disables the gallery/documents button if there are no common images/files. It seems to be missing here. This way we don't need an extra string saying "the gallery is empty" or something like that.

I looked into this: I will add an empty state label for consistency. 

> * Some images are not displayed correctly in the grid view. It looks like there's a dummy image in the background, and if there's an image with transparency, you can see it in the background.
>   Also some images are not displayed at all

Hopefully fixed. :)

>   [EDIT: The dummy image in the background is not true. There captured image has exactly that table view cells in the background :P. That other images are not displayed at all is still true though.]
> 
> I will attach an example:
> ![gallery1](https://user-images.githubusercontent.com/2614900/83004335-d5320480-a00f-11ea-8509-3ed4f2e12ff1.png)
> 
> the image that is not shown in the gallery from above example
> ![Screen Shot 2020-05-27 at 11 41 25](https://user-images.githubusercontent.com/2614900/83004408-ec70f200-a00f-11ea-8ca9-7fcda84cc835.png)
> 
> * Videos are also displayed incorrectly. Instead of an thumbnail, the dummy image from the axample above is shown. I think either videos should be excluded here for now or shown as a thumbnail image.

Fixed by showing thumbnails.

> * On Android the gallery shows also gifs. Since we now use the SDWebImage framework, it can probably easily archieved on iOS. But this can also be part of another PR.

Fixed.